native/acter/src/api/account/three_pid.rs

```diff
@@ -93,27 +145,90 @@ impl ThreePidManager {
                 };
                 account.set_account_data(content).await?;
 
+                Ok(ThreePidEmailTokenResponse {
+                    inner,
+                    client_secret,
+                })
+            })
+            .await?
+    }
+
+    // add 3pid on the homeserver for this account
+    // this 3pid may be used by the homeserver to authenticate the user during sensitive operations
+    #[cfg(feature = "testing")]
+    pub async fn add_3pid(
+        &self,
+        client_secret: String,
+        sid: String,
+        password: String,
+    ) -> Result<bool> {
+        let client = self.client.deref().clone();
+        let account = self.account.clone();
+        let user_id = self.user_id.clone();
+        let client_secret = ClientSecret::parse(&client_secret)?;
+        let sid = SessionId::parse(&sid)?; // it was already related with email or msisdn
+
+        RUNTIME
+            .spawn(async move {
+                let capabilities = client.get_capabilities().await?;
+                if !capabilities.thirdparty_id_changes.enabled {
+                    bail!("Server doesn't support 3pid change");
+                }
+                if let Err(e) = account.add_3pid(&client_secret, &sid, None).await {
+                    let Some(inf) = e.as_uiaa_response() else {
+                        return Err(clearify_error(e));
+                    };
+                    let pswd = assign!(Password::new(user_id.into(), password), {
+                        session: inf.session.clone(),
+                    });
+                    let auth_data = AuthData::Password(pswd);
+                    account
+                        .add_3pid(&client_secret, &sid, Some(auth_data))
+                        .await
+                        .map_err(clearify_error)?;
+                }
                 Ok(true)
             })
             .await?
     }
 
+    pub async fn get_3pids(&self, address: String) -> Result<Vec<ThreePid>> {
```

Comment:
The parameter is not used afaics. And to be in line with rust styles let's drop teh `get_`-prefix. I'd suggest renaming that entire thing (as mentioned above) as well. How about:

```suggestion
    pub async fn externalIds(&self) -> Result<Vec<ThreePid>> {
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1896/commits/94a92a07b2cc1729b30fea64158a3ae92a586328 and https://github.com/acterglobal/a3/pull/1896/commits/b6f0caa06e3e704a674318942030b2aedf89d02a